### ISSUE TITLE
Podfile does not exist and won't get created

### ISSUE THREAD

**Issue by raenk** [flutter/flutter#131427]
I've been looking for a solution and haven't found any for my circumstances.

I develop in linux using VSCode, building and deploying to the AppStore via codemagic.

I have a published App, both stores (apple/android) and just recently updated its dependencies and added these new dependencies:

```
webview_flutter: ^4.2.2
package_info_plus: ^4.0.2
```

PlayStore: Built without trouble for android and the updated build was published. 
AppStore: Went to codemagic to build/deploy it and it failed. The error:

`Automatically assigning platform 'iOS' with version '11.0' on target 'Runner' because no platform was specified. Please specify a platform for this target in your Podfile.`

Say no more. I go to look for the Podfile... missing!
Nobody deleted it (or any other pod related data) and the previous built went smooth! Oh well, maybe something changed (new requirements,etc).

Went and tried to recreate the project just for iOS with:
```flutter create --platforms ios```

Command executes fine, but still no Podfile/pod stuff!

All "solutions" I have found suggests actions that are just for a Mac environment (run pod commands). Not my case.

Basically, I have reached a dead end and it's driving me crazy.

What am i missing?

**Comment by darshankawar** [comment]
> just recently updated its dependencies and added these new dependencies:
> 
> ```
> webview_flutter: ^4.2.2
> package_info_plus: ^4.0.2
> ```
> 
> PlayStore: Built without trouble for android and the updated build was published. AppStore: Went to codemagic to build/deploy it and it failed. The error:
> 
> `Automatically assigning platform 'iOS' with version '11.0' on target 'Runner' because no platform was specified. Please specify a platform for this target in your Podfile.`

@raenk This error could be occuring due to the fact that the plugins you added may require a minimum OS version higher than the app. 
Can you provide the complete error message ? Do you get something like `dependency were found, but they required a higher minimum deployment target.` ?

Usually the podfile is generated if you add any plugin to a project and if you are creating a new barebone project, it won't generate the podfile. Also, in your case the podfile could be missing because you added a third party package `package_info_plus` that you can check and confirm.

**Comment by raenk** [comment]
Thanks @darshankawar I'm not quite following you about the reasons or circumstances the Podfile should or should not be created.

This is the full error log:
```
== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/27-07-23_mpl4myzg.keychain-db
Searching for files matching /Users/builder/clone/ios/**/*.xcodeproj
Completed configuring code signing settings
 - Using profile "AdKondo ios_app_store 1674591911" [3c8b06b6-6bf7-4039-ab83-a9dee55c62c5] for target "Runner" [Debug] from project "Runner"
 - Using profile "AdKondo ios_app_store 1674591911" [3c8b06b6-6bf7-4039-ab83-a9dee55c62c5] for target "Runner" [Profile] from project "Runner"
 - Using profile "AdKondo ios_app_store 1674591911" [3c8b06b6-6bf7-4039-ab83-a9dee55c62c5] for target "Runner" [Release] from project "Runner"
Generated options for exporting the project
 - Method: app-store
 - Provisioning Profiles:
     - com.adkondo.app: AdKondo ios_app_store 1674591911
 - Signing Certificate: Apple Distribution
 - Signing Style: manual
 - Team Id: ZJWLWG6MMP
Saved export options to /Users/builder/export_options.plist

> flutter build ipa --release --export-options-plist /Users/builder/export_options.plist
Resolving dependencies...
  collection 1.17.1 (1.18.0 available)
  dart_internal 0.2.8 (0.2.9 available)
  file 6.1.4 (7.0.0 available)
  intl 0.18.0 (0.18.1 available)
  matcher 0.12.15 (0.12.16 available)
  material_color_utilities 0.2.0 (0.8.0 available)
  plugin_platform_interface 2.1.4 (2.1.5 available)
  source_span 1.9.1 (1.10.0 available)
  stack_trace 1.11.0 (1.11.1 available)
  stream_channel 2.1.1 (2.1.2 available)
  test_api 0.5.1 (0.6.1 available)
  url_launcher_android 6.0.36 (6.0.37 available)
  webview_flutter_android 3.9.1 (3.9.2 available)
  webview_flutter_wkwebview 3.7.0 (3.7.2 available)
  win32 5.0.5 (5.0.6 available)
Got dependencies!

Archiving com.adkondo.app...
Updating minimum iOS deployment target to 11.0.
Upgrading project.pbxproj
Upgrading AppFrameworkInfo.plist
Upgrading Podfile
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Upgrading Info.plist
Removing script build phase dependency analysis.
Adding input path to Thin Binary build phase.
Automatically signing iOS for device deployment using specified development team in Xcode project: ZJWLWG6MMP
Running pod install...                                             429ms
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
    [!] Unable to find a target named `RunnerTests` in project `Runner.xcodeproj`, did find `Runner`.

    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer/target_inspector.rb:108:in `compute_targets'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer/target_inspector.rb:40:in `compute_results'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1197:in `block (3 levels) in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1195:in `each'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1195:in `block (2 levels) in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1193:in `each'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1193:in `block in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1188:in `inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:107:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:422:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:162:in `install!'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command/install.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/bin/pod:55:in `<top (required)>'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `load'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `<main>'

Error output from CocoaPods:
↳

    [!] Automatically assigning platform `iOS` with version `11.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

Error running pod install


Build failed :|
Failed to build for iOS
```

**Comment by darshankawar** [comment]
@raenk 
Looking at the error log, there's another warning `Unable to find a target named RunnerTests in project Runner.xcodeproj, did find Runner.` which could be leading to the error. Have you tried to resolve it ?

As I mentioned earlier, if you just use `webview_flutter` for example and try to run, do you still get same error ?

**Comment by raenk** [comment]
Solved following these steps suggested at the discussion forum of codemagic:

1. Start a build, enable SSH connection
2. SSH or VNC into the machine
3. When in the machine, navigate to /Users/builder/clone/ios
4. There, a Podfile should be generated. Copy the contents of this Podfile, and create a Podfile locally inside your iosdirectory.
5. Paste the contents of the VM-s Podfile to your locally created one.
6. Now, inside the Podfile, find this part and comment out these lines which I've marked:

```
target 'Runner' do
 use_frameworks!
 use_modular_headers!

 flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
 # target 'RunnerTests' do <-- MAKE SURE THIS LINE IS COMMENTED OUT
  # inherit! :search_paths <-- MAKE SURE THIS LINE IS COMMENTED OUT
 # end <-- MAKE SURE THIS LINE IS COMMENTED OUT
end
```

7. Commit changes
8. Run a build

**Comment by darshankawar** [comment]
Does that mean the issue is resolved ?

**Comment by raenk** [comment]
Well, that solution worked, but not sure if flutter is doing something wrong with the iOS folder.

**Comment by darshankawar** [comment]
Ok, if you have concrete steps to replicate the issue wherein it says the podfile is missing, please provide us the details here so that we can make this issue actionable.

**Comment by raenk** [comment]
Well, the steps are in my original post. Someone at codemagic community suggested that it could be due to creating my project in an older version of flutter (first build published) and building an update with the latest flutter version.

**Comment by darshankawar** [comment]
I tried by creating new flutter project with a recent and latest version but didn't get any issues. Maybe this could be a config issue at your end unless you are able to replicate it locally using older version of flutter.

**Comment by DizzasTeR** [comment]
I followed the same instructions for codemagic as listed by @raenk to fix my build pipeline for iOS

It is worth noting that this issue mainly happens to people who don't own a mac and therefore cannot use XCode to get flutter to fix podfile issues
